packages/ui-extender/src/actions/menuAction/menuAction.tsx

```diff
@@ -359,22 +360,22 @@ export type MenuActionComponentProps = {
     /**
      * Root menu context, if sub menu (internal)
      */
-    rootMenuContext?: MenuContext,
+    rootMenuContext: MenuContext,
 
     /**
      * Parent menu context, if sub menu (internal)
      */
-    parentMenuContext?: MenuContext,
+    parentMenuContext: MenuContext,
```

Comment:
These look like they should be optional based on code usage

Reply:
<img width="1846" height="1508" alt="Image" src="https://github.com/user-attachments/assets/e1ed7f9e-8b6f-4811-a432-1aff714d4c83" />

[We don't seem to consider them optional in the code,](https://github.com/Jahia/javascript-components/blob/cc296cea54078923f00699b1657c5f60833a81d1/packages/ui-extender/src/actions/menuAction/menuAction.tsx#L79-L111) but I don't understand the full abstraction, so there might be something I'm missing

Reply:
I fully agree with you, but it's a game of pulleys and levers I can't solve: making them optional has consequences in `packages/ui-extender/src/actions/menuAction/ContextualMenu.tsx`

<img width="1060" height="340" alt="image" src="https://github.com/user-attachments/assets/2761fcb9-edbd-4ba5-8c30-0a2c14c1c87a" />

This `this.onClickRef.current = onClick` line binds `ItemRenderProps` and `MenuActionComponentProps`

You should checkout the code to see how unsound the current implementation is 👀

Reply:
Had discussion with @GauBen and convinced me to trust the compiler and go with required for now. 🙏